src/main/java/org/broadinstitute/dsm/jobs/TestBostonUPSTrackingJob.java

```diff
@@ -508,6 +543,43 @@ private void updateTrackingInfo(PoolingDataSource<PoolableConnection> dataSource
 
     }
 
+    /**
+     * Queries the database to see whether there is an upload reason
+     * for the given kit label
+     */
+    private boolean doesKitHaveUploadReason(Connection conn, String kitLabel) throws SQLException {
+        boolean hasUploadReason = false;
+        try (PreparedStatement stmt = conn.prepareStatement(SQL_GET_KIT_REQUEST + BY_KIT_LABEL)) {
```

Comment:
I don't think this query will work since that is only selecting from `ddp_kit_request` while `ddp_kit` has the `kit_label`. I'm afraid you have to add a query of their left join on `dsm_kit_request_id`

Reply:
Oops!  Ok, will see if there's a different reusable snippet of SQL here and if not, will roll one as you suggest.